### PR TITLE
HW3_v1

### PR DESCRIPTION
**Pull request by NikonovaEkaterina**
HW3 version 1

### CODE REVIEW

File: src/test/java/hw3/data/GetProperties.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            PROPERTIES = new Properties();
+            PROPERTIES.load(fileInputStream);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            if (fileInputStream != null)
+                try {
+                    fileInputStream.close();
+                } catch (IOException e) {
+                    e.printStackTrace(); } } }
```
Comment (komyagin) [inline src/test/java/hw3/data/GetProperties.java]: Ctrl+Alt+L formatting the code in the IntelliJ IDEA. It's not according to Java convention to close } in one line

File: src/test/java/hw3/page/LoginPage.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+       loginButton.click();
+   }
+
+// check user name
+    public void checkUserName(String text)
+    {
+
+        assertEquals(userLogName.getText(), text);
+    }
```
Comment (komyagin) [inline src/test/java/hw3/page/LoginPage.java]: Ctrl+Alt+L formatting the code in the IntelliJ IDEA.

File: src/test/java/hw3/page/DefaultPageImages.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public WebElement iconBase;
+
+    //get all images amount
+    @FindBy(css="ul.uui-navigation.nav.navbar-nav.m-l8>li")
+    public List<WebElement> images;
+
+    //check images amount
+    public void checkImagesAmount(Integer num)
+    {
+        Integer size =images.size();
```
Comment (komyagin) [inline src/test/java/hw3/page/DefaultPageImages.java]: Why do you use here an Integer, but not an int? And there is no space after the equals sign.

File: src/test/java/hw3/data/AbstractPageTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+public class AbstractPageTest {
+
+
+    protected static WebDriver driver;
```
Comment (komyagin) [inline src/test/java/hw3/data/AbstractPageTest.java]: why do you decide to use static?

File: src/test/java/hw3/ex1/DefaultPageVPOTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        image = new DefaultPageImages(driver);
+        imageText = new DefaultPageImagesText(driver);
+        frame = new DefaultPageFrames(driver);
+        leftMenuItem = new LeftMenuItems(driver);
+        multiText = "Already have good base\n" +
+                "(about 20 internal and\n" +
+                "some external projects),\n" +
+                "wish to get more…";
+
+    }
```
Comment (komyagin) [inline src/test/java/hw3/ex1/DefaultPageVPOTest.java]: why do you decide to use static?

File: src/test/java/hw3/data/AbstractPageTest.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        driver.manage().window().maximize();
+    }
+
+
+    @AfterClass(alwaysRun = true)
+    public  void ChromeClose()
+    {
+        driver.quit();
+        driver=null;
+    }
```
Comment (komyagin) [inline src/test/java/hw3/data/AbstractPageTest.java]: It's a C# style to open brackets like this.

File: src/test/java/hw3/page/DefaultPageFrames.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    @FindBy(id = "frame-button")
+    public WebElement iframeButton;
+
+
+    //check frame button
+
+    public void checkButton()
+    {
+        iframeButton.isDisplayed();
+    }
```
Comment (komyagin) [inline src/test/java/hw3/page/DefaultPageFrames.java]: It's a C# style to open brackets like this.

File: src/test/java/hw3/page/DefaultPageImages.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    @FindBy(css="div.benefit-icon>span.icons-benefit.icon-base")
+    public WebElement iconBase;
+
+    //get all images amount
+    @FindBy(css="ul.uui-navigation.nav.navbar-nav.m-l8>li")
+    public List<WebElement> images;
+
+    //check images amount
+    public void checkImagesAmount(Integer num)
+    {
```
Comment (komyagin) [inline src/test/java/hw3/page/DefaultPageImages.java]: It's a C# style to open brackets like this.

File: src/test/java/hw3/page/DefaultPageImagesText.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        Integer size =imagesText.size();
+        assertEquals(size, num);
+    }
+
+    //check images text presence
+    public void imageTextIsDisplayed(WebElement element)
+    {
+        element.isDisplayed();
+    }
```
Comment (komyagin) [inline src/test/java/hw3/page/DefaultPageImagesText.java]: It's a C# style to open brackets like this.

File: src/test/java/hw3/page/LoginPage.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+   public void clickLoginButton()
+   {
+
+       loginButton.click();
+   }
+
+// check user name
+    public void checkUserName(String text)
+    {
+
```
Comment (komyagin) [inline src/test/java/hw3/page/LoginPage.java]: It's a C# style to open brackets like this.

**Review by komyagin — CHANGES_REQUESTED** [review]

File: src/test/java/hw3/data/GetProperties.java (hunk 1 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/data/GetProperties.java]: code format updated

File: src/test/java/hw3/page/LoginPage.java (hunk 2 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/page/LoginPage.java]: code format updated

File: src/test/java/hw3/page/DefaultPageImages.java (hunk 3 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/page/DefaultPageImages.java]: changed to int

File: src/test/java/hw3/data/AbstractPageTest.java (hunk 4 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/data/AbstractPageTest.java]: removed static modificator

File: src/test/java/hw3/ex1/DefaultPageVPOTest.java (hunk 5 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/ex1/DefaultPageVPOTest.java]: removed static modifacator

File: src/test/java/hw3/data/AbstractPageTest.java (hunk 6 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/data/AbstractPageTest.java]: code format updated

File: src/test/java/hw3/page/DefaultPageFrames.java (hunk 7 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/page/DefaultPageFrames.java]: code format updated

File: src/test/java/hw3/page/DefaultPageImagesText.java (hunk 9 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/page/DefaultPageImagesText.java]: code format updated

File: src/test/java/hw3/page/LoginPage.java (hunk 10 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw3/page/LoginPage.java]: code format updated

**Review by komyagin — APPROVED** [review]

**Comment by komyagin** [comment]
HW3 is accepted

**Pull request merged**